liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -72,43 +70,43 @@ static synchronized LocationEngine getLocationEngine(Context context) {
 
   @Override
   public void activate() {
-    connect();
-  }
-
-  @Override
-  public void deactivate() {
-    if (googleApiClient != null && googleApiClient.isConnected()) {
-      googleApiClient.disconnect();
+    for (LocationEngineListener listener : locationListeners) {
+      listener.onConnected();
     }
   }
 
   @Override
-  public boolean isConnected() {
-    return googleApiClient.isConnected();
-  }
+  public void deactivate() {
 
-  @Override
-  public void onConnected(@Nullable Bundle bundle) {
-    for (LocationEngineListener listener : locationListeners) {
-      listener.onConnected();
-    }
   }
 
   @Override
-  public void onConnectionSuspended(int cause) {
+  public boolean isConnected() {
+    return true;
   }
 
   @Override
-  public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+  public void requestLastLocation(final LastLocationListener lastLocationListener) {
+    //noinspection MissingPermission
+    fusedLocationProviderClient.getLastLocation()
+      .addOnSuccessListener(new OnSuccessListener<Location>() {
+        @Override
+        public void onSuccess(Location location) {
+          lastLocationListener.onLastLocationSuccess(location);
+        }
+      })
+      .addOnFailureListener(new OnFailureListener() {
+        @Override
+        public void onFailure(@NonNull Exception exception) {
+          lastLocationListener.onLastLocationFail(exception);
+        }
+      });
   }
 
+  @Deprecated
   @Override
   public Location getLastLocation() {
-    if (googleApiClient.isConnected()) {
-      //noinspection MissingPermission
-      return LocationServices.FusedLocationApi.getLastLocation(googleApiClient);
-    }
-
+    // No op
```

Comment:
"Deprecating" doesn't mean to remove working functionality. What about using `fusedLocationProviderClient.getLastLocation().getResult()`? I believe that if connected will return the last location, `null` otherwise.